lib/param_parsing/versiontf.go

```diff
@@ -0,0 +1,26 @@
+package param_parsing
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/warrensbox/terraform-switcher/lib"
+	"log"
+)
+
+func GetVersionFromVersionsTF(params Params) Params {
+	filePath := params.ChDirPath + "/version.tf"
+	if lib.CheckFileExist(filePath) {
+		fmt.Printf("Reading version from %s\n", filePath)
+		module, err := tfconfig.LoadModule(params.ChDirPath)
+		if err != nil {
+			log.Fatal("Could not load terraform module")
+		}
+		tfconstraint := module.RequiredCore[0]
```

Comment:
As there may be multiple version constraint defined (I guess this is allowed by TF), we may consider using not just the first one, but append all the constraint items split by comma to the var value.

Reply:
You mean multiple version constraints via possible sub modules?
Terraform does not allow more than one terraform version constraint per module afaik.
So it would always be module.RequiredCore[0]

Reply:
The previous was run using `1.3.10`.
And when I switch to e.g. `1.5.7`, I get this errors:
```
> terraform validate
╷
│ Error: Unsupported Terraform Core version
│
│   on main.tf line 2, in terraform:
│    2:   required_version = "~> 1.3.6"
│
│ This configuration does not support Terraform version 1.5.7. To proceed, either choose another supported Terraform
│ version or update this version constraint. Version constraints are normally set for good reason, so updating the
│ constraint may lead to other errors or unexpected behavior.
╵
╷
│ Error: Unsupported Terraform Core version
│
│   on main.tf line 12, in terraform:
│   12:   required_version = "<= 1.4.0"
│
│ This configuration does not support Terraform version 1.5.7. To proceed, either choose another supported Terraform
│ version or update this version constraint. Version constraints are normally set for good reason, so updating the
│ constraint may lead to other errors or unexpected behavior.
```

Reply:
Somewhere around this file is where they collect all constraints from the entire module: https://github.com/hashicorp/terraform/blob/main/internal/configs/module.go#L205

Reply:
Or here is the better place I guess: https://github.com/hashicorp/terraform-schema/blob/main/earlydecoder/load_module.go#L71

Reply:
I was not aware that this is possible. We should definitely implement that. I see that as an extra issue though as i do not intend to implement behaviour changes with this ticket. I just wanted to make sure that the in the readme described behaviour is working.
This PR is large enough as it is. 😬 

So please let's create another issue for that

Reply:
Done: https://github.com/warrensbox/terraform-switcher/issues/357
Please feel free to adjust description if and as needed.